src/sync/remote_mongo_collection.cpp

```diff
@@ -18,3 +18,505 @@
+                                         std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        try {
+            auto json = nlohmann::json::parse(value.value());
+            auto count_string = json.at("deletedCount").at("$numberInt").get<std::string>();
+            return completion_block(std::stoi(count_string), error);
+        } catch (const std::exception& e) {
+            return completion_block(0, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+        }
+    }
+    
+    return completion_block(0, error);
+}
+
+static void handle_update_response(util::Optional<AppError> error,
+                                   util::Optional<std::string> value,
+                                   std::function<void(RemoteMongoCollection::RemoteUpdateResult, util::Optional<AppError>)> completion_block)
+{
+    if (value && !error) {
+        
+        try {
+            auto json = nlohmann::json::parse(value.value());
+            auto matched_count_string = json.at("matchedCount").at("$numberInt").get<std::string>();
+            auto matched_count = std::stoull(matched_count_string);
+            auto modified_count_string = json.at("modifiedCount").at("$numberInt").get<std::string>();
+            auto modified_count = std::stoull(modified_count_string);
+            std::string upserted_id;
+            auto it = json.find("upsertedId");
+            if (it != json.end()) {
+                upserted_id = json.at("upsertedId").at("$oid").get<std::string>();
+            }
+            
+            return completion_block(RemoteMongoCollection::RemoteUpdateResult {
+                matched_count,
+                modified_count,
+                upserted_id
+            }, error);
+        } catch (const std::exception& e) {
+            return completion_block({}, AppError(make_error_code(JSONErrorCode::malformed_json), e.what()));
+        }
+    }
+    
+    return completion_block({}, error);
+}
+
+void RemoteMongoCollection::find(const std::string& filter_json,
+                                 RemoteFindOptions options,
+                                 std::function<void(std::string, util::Optional<AppError>)> completion_block)
+{
+    try {
+        auto base_args = m_base_operation_args;
+        base_args.push_back({ "query", nlohmann::json::parse(filter_json) });
+        
+        if (options.limit) {
+            base_args.push_back({ "limit", options.limit.value() });
+        }
+        
+        if (options.projection_json) {
+            base_args.push_back({ "project", nlohmann::json::parse(options.projection_json.value()) });
+        }
+        
+        if (options.sort_json) {
+            base_args.push_back({ "sort", nlohmann::json::parse(options.sort_json.value()) });
+        }
+        
+        auto args = nlohmann::json( {{"arguments", nlohmann::json::array({base_args} ) }} );
+                
+        m_service->call_function("find",
```

Comment:
I'm having trouble finding the docs for this. I'm wondering how this works for large collections because I think you'd typically get a cursor back - do we have to handle that or is this actually returning everything?

Reply:
we don't have to handle the cursor, the find will return everything, or if a filter is passed, all documents matching the filter